src/cv/saliency.cpp

```diff
@@ -1,3 +1,142 @@
 #include "cv/saliency.hpp"
+#include <torch/script.h> // One-stop header.
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include <memory>
+#include "cv/saliency.hpp"
+#include <torch/cuda.h>
+#include <torch/mps.h>
+#include "utilities/logging.hpp"
+
+// constructor definition
+Saliency::Saliency () {
+    this->modelPath = "";
+} 
+
+// constructor definition
+Saliency::Saliency (std::string modelPath) {
+    this->modelPath = modelPath;
+} 
+
+std::vector<CroppedTarget> Saliency::salience(cv::Mat image) { 
+    
+    at::Tensor tensor = Saliency::ToTensor(image, false, false, 0);
+
+    // convert the tensor into float and scale it 
+    tensor = tensor.toType(c10::kFloat).div(255);
+    
+    // swap axis 
+    tensor = Saliency::transpose(tensor, { (2),(0),(1) });
+
+    auto input_to_net = ToInput(tensor);
+
+    c10::Device device = torch::cuda::is_available() ? torch::kCUDA : torch::kCPU;
+    torch::jit::script::Module module;
+    try {
+        // Deserialize the ScriptModule from a file using torch::jit::load().
+        module = torch::jit::load(modelPath, device);
+        module.eval();
+    }
+    catch (const c10::Error& e) {
+        std::cerr << "error loading the model\n";
+        std::cerr << e.msg() << std::endl; 
+        return {};
+    }
+
+    // note: add flag so that forward () runs using CUDA on the jetson
+        
+    auto output = module.forward(input_to_net); // vector of IValues which contains c10::List to IValue output
+    c10::ivalue::Tuple& tuple = output.toTupleRef();
+    auto detections = tuple.elements()[1];      // index 1: detections
+    std::vector<CroppedTarget> targets;
+    auto listDetections = detections.toList();
+
+    for (c10::IValue results: listDetections) {
+        c10::Dict<c10::IValue, c10::IValue> dict = results.toGenericDict();     // dict whose fields are boxes, labels, scores. 
+                                                        // see: https://pytorch.org/vision/main/models/generated/torchvision.models.detection.fasterrcnn_resnet50_fpn.html#torchvision.models.detection.fasterrcnn_resnet50_fpn
+        c10::IValue boxes = dict.at("boxes");           // https://discuss.pytorch.org/t/genericdict-c/178178/3
+        c10::IValue labels = dict.at("labels");          
+        at::Tensor boxTensor = boxes.toTensor();
+        at::Tensor labelsTensor = labels.toTensor();
+        // std::cout << boxTensor << std::endl;
+        // std::cout << labelsTensor << std::endl;
+        
+        for (int i = 0; i < boxTensor.size(0); i++) { 
+            CroppedTarget targ; 
```

Comment:
I don't like nonstandard abbreviations like this. Please feel free to have descriptive variable names.